### PR TITLE
AgentBusinessUtr and validation for AtedUtr

### PR DESCRIPTION
**Pull request by nonimay**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/domain/AtedUtr.scala [hunk 1]
```diff
@@ -27,4 +27,21 @@ case class AtedUtr(utr: String) extends TaxIdentifier with SimpleName {
...
   implicit val atedUtrWrite: Writes[AtedUtr] = new SimpleObjectWrites[AtedUtr](_.value)
   implicit val atedUtrRead: Reads[AtedUtr] = new SimpleObjectReads[AtedUtr]("utr", AtedUtr.apply)
+
+  private val validFormat = "^[Xx][a-zA-Z]\\d{2}00000\\d{6}$"
+
+  def isValid(utr: String) = !utr.isEmpty && utr.matches(validFormat) && isCheckCorrect(utr.toUpperCase)
+
+  private def isCheckCorrect(utr: String): Boolean = utr.charAt(1) == getCheckCharacter(utr)
+
+  private val checkString = "ABCDEFGHXJKLMNYPQRSTZVW"
```
Comment (howyp) [inline src/main/scala/uk/gov/hmrc/domain/AtedUtr.scala]: This seems very similar to the code in `object AgentBusinessUtr`

Comment (nonimay) [inline src/main/scala/uk/gov/hmrc/domain/AtedUtr.scala]: it is... Ive just been refactoring this code

File: src/main/scala/uk/gov/hmrc/domain/AgentBusinessUtr.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.domain
+
+import play.api.libs.json.{Reads, Writes}
+
+case class AgentBusinessUtr(utr: String) extends TaxIdentifier with SimpleName {
+  override lazy val toString = utr
```
Comment (howyp) [inline src/main/scala/uk/gov/hmrc/domain/AgentBusinessUtr.scala:23]: There's no `require` here, so I can create invalid `AgentBusinessUtr` objects with invalid UTRs - doesn't that defeat the purpose of the type?

File: src/main/scala/uk/gov/hmrc/domain/CheckCharacter.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.domain
+
+trait CheckCharacter {
+
+  protected val checkString = "ABCDEFGHXJKLMNYPQRSTZVW"
+  protected val mod = 23
+  protected val weights = List(9, 10, 11, 12, 13, 8, 7, 6, 5, 4, 3, 2, 1)
```
Comment (howyp) [inline src/main/scala/uk/gov/hmrc/domain/CheckCharacter.scala:23]: Couldn't we do the `zipWithIndex` here?

File: src/main/scala/uk/gov/hmrc/domain/AgentBusinessUtr.scala (hunk 2 above)
Comment (howyp) [inline src/main/scala/uk/gov/hmrc/domain/AgentBusinessUtr.scala:23]: I'd have the same comment about `AtedUtr`

Comment (nonimay) [inline src/main/scala/uk/gov/hmrc/domain/AgentBusinessUtr.scala:23]: so there isn't...

**Pull request merged**